apps/vmq_webhooks/src/vmq_webhooks_plugin.erl

```diff
@@ -751,11 +751,34 @@ normalize_sub_topics(Mods, _Opts) ->
     lists:map(
       fun({topics, Topics}) ->
               {topics,
-               [{proplists:get_value(<<"topic">>, T),
-                 proplists:get_value(<<"qos">>, T)} || T <- Topics]};
+               lists:map(
+                 fun(T) ->
+                         Topic = proplists:get_value(<<"topic">>, T),
+                         QoS = proplists:get_value(<<"qos">>, T),
+                         case lists:member(QoS, [0,1,2]) of
+                             true ->
+                                 NL = proplists:get_value(<<"no_local">>, T, false),
+                                 RAP = proplists:get_value(<<"rap">>, T, false),
+                                 RH = retain_handling(
+                                        proplists:get_value(<<"retain_handling">>, T, send_retain)),
+                                 {Topic, {QoS, #{no_local => NL,
+                                                 rap => RAP,
+                                                 retain_handling => RH}}};
+                             _ ->
+                                 {Topic, QoS}
+                         end
+                 end, Topics)};
          (E) -> E
       end, Mods).
 
+retain_handling(<<"send_retain">>) -> send_retain;
+retain_handling(<<"send_if_new_sub">>) -> send_if_new_sub;
+retain_handling(<<"dont_send">>) -> dont_send;
+retain_handling(Val) when is_atom(Val) ->
+    Val;
+retain_handling(Val) ->
+    throw({invalid_retain_handling, Val}).
```

Comment:
@dergraf this isn't my favorite what of handling errors, but thought I'd be better to throw something with a meaning than just plain crashing.